### DISCUSSION TITLE
Repository APIs seem to be missing point queries for invitations

### DISCUSSION THREAD

**Discussion by jeffwilcox** [community/community#69417] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

Trying to improve the way that we handle Outside Collaborator invitations and experiences, I've been scratching my head on how to manage the invitation APIs.

- There's the API to list all invites for a repository, which can get long in some scenarios, and so can be slow to page through.
- There's an API to check a login's permissions on a repo (https://docs.github.com/en/rest/collaborators/collaborators?apiVersion=2022-11-28#check-if-a-user-is-a-repository-collaborator); however, it doesn't return something like an HTTP code that would indicate if someone has a pending invite
- The API to check the permission level for a user returns `none` for an invited user and no invitation ID or other data (https://docs.github.com/en/rest/collaborators/collaborators?apiVersion=2022-11-28#get-repository-permissions-for-a-user)
- There is no invitation API to get/check for an invitation in a repo for a single user (docs page https://docs.github.com/en/rest/collaborators/invitations?apiVersion=2022-11-28)
- No GraphQL data on repository invitations

The scenario I'm missing is just a point query: given my repo, and a login, I want to either get that login's invitation state (invitation ID, invited date, etc.), or, a 404.

**Comment by frankie567** [comment]
I would also be interested in having a `GET` endpoint to retrieve a single invitation for a given user and repository.

It can be useful in scenarios where you need to update a permission: update the invitation if it's not yet accepted or update the permission of the existing collaborator.
